backend/routes/friends.js

```diff
@@ -15,13 +15,21 @@ router.get('/users', async (req, res) => {
         if (!userExistsCheck) {
             res.status(200).send({})
         }
+    } catch {
+        res.status(400).send({"message": "Parse Query failed"})
+    }
+    try {
```

Comment:
thank you for addressing the feedback and even adding in more detailed handling for each failure, it really shows an in-depth understanding of the code.